modules/helm/crds.mk

```diff
@@ -37,29 +37,29 @@ ifeq ($(HOST_OS),darwin)
 	sed_inplace := sed -i ''
 endif
 
+crds_dir ?= config/crd/bases
```

Comment:
I think it could make sense to have a proper default for this path. Alternatives:

1. deploy/crds ([cert-manager](https://github.com/cert-manager/cert-manager/tree/master/deploy/crds))
2. test/testdata (suggested in https://github.com/cert-manager/trust-manager/pull/482)

Reply:
I actually quite like the idea of using `test/...`. One of the concerns around this was that we don't want people to think that these CRDs are a blessed thing they can use for deployments. In cert-manager we use deploy/crds because it works a bit differently, but for other projects I think `test/...` would signal to people that we don't necc. support them downloading these checked-in CRDs and using them.

My only suggestion there would be that for `testdata` I'd be expecting files, like [here](https://github.com/cert-manager/cert-manager/tree/e9ca5ee823952e2c8a83c2e3eddf0bb31c454fd2/internal/pem).

Maybe `test/crds`?

Ultimately it doesn't matter much though, I don't feel too strongly. My preferences in order would be:

1. `test/crds`
2. `test/testdata`
3. `deploy/crds`
4. Anywhere else

Reply:
Option 1 `deploy/crds` makes sense to me assuming this is where the RAW non-helm'd CRDs will go?

Reply:
As @SgtCoDFish is stating, cert-manager is a bit special in this area. It is using schemapatch, and not generate controller-gen command, probably for "historical" reasons. So I would vote for staying away from deploy/crds per default.

Putting it under "test" feels a bit wrong to me, since these files will be used as input to the Helm charts. But absolutely no blocker for me. 😊

Reply:
Per https://github.com/cert-manager/trust-manager/pull/475#discussion_r1858035321, I think we should make room for also checking in generating webhook configuration resources (and RBAC roles and clusterroles).

Reply:
> Putting it under "test" feels a bit wrong to me, since these files will be used as input to the Helm charts. But absolutely no blocker for me. 😊

Ah, I'm not sure I'd realised that they would be used as inputs for the helm charts. In that case, something like `deploy/crds` starts to make more sense.

If we want to discourage use, something like `deploy/internal/crds`?

Reply:
The default path is now to changed to `deploy/crds`.